lib/services/jellyfin_api_helper.dart

```diff
@@ -865,31 +866,59 @@ class JellyfinApiHelper {
     return uri;
   }
 
-  CanDeleteFromServerResponse canDeleteFromServer(BaseItemDto item) {
-    // Cant delete from server when offline anyway
-    if (FinampSettingsHelper.finampSettings.isOffline) {
-      return CanDeleteFromServerResponse(false, null);
+  late final ProviderFamily<bool, CanDeleteRequest>
+      canDeleteFromServerProvider =
+      ProviderFamily((ref, CanDeleteRequest item) {
+    bool offline = ref.watch(
+        finampSettingsProvider.select((value) => value.requireValue.isOffline));
+    if (offline) {
+      return false;
     }
-
-    // Cant delete if setting is disabled anyway
-    if (!FinampSettingsHelper.finampSettings.allowDeleteFromServer) {
-      return CanDeleteFromServerResponse(false, null);
+    bool deleteEnabled = ref.watch(finampSettingsProvider
+        .select((value) => value.requireValue.allowDeleteFromServer));
+    if (!deleteEnabled) {
+      return false;
     }
+    // do not bother checking server for item types known to not be deletable
+    var itemType = BaseItemDtoType.fromItem(item.item);
+    if (itemType != BaseItemDtoType.album &&
+        itemType != BaseItemDtoType.playlist &&
+        itemType != BaseItemDtoType.track) {
+      return false;
+    }
+    bool? serverReturn =
+        ref.watch(_canDeleteFromServerAsyncProvider(item.item.id)).value;
+    if (serverReturn == null) {
+      // fallback to true in case the response is invalid but the user could delete still
+      // worst case would be an error messing when trying to delete
```

Comment:
```
      // fallback to allowing deletion even if the response is invalid, since the user might still be able to delete
      // worst case would be getting an error message when trying to delete
```
just a typo fix and clearer wording

Reply:
Although I'm not sure if this would ever happen, since errors / invalid responses are handled by the `catchError()` handler of the `_canDeleteFromServerAsyncProvider`...